---
title: "sex_comp_align_analysis"
output: html_document
date: "2024-06-24"
---
```{r}
library(dplyr)
library(stringr)
library(DESeq2)
library(corrplot)
library(ggplot2)
library(ggfortify) 
unfiltered_glowing_counts <- read.csv("unfiltered_glowing_counts.csv")
load("log_PFAS_pheno.Rdata")
```

#creating sex completement and default alignment datasets
```{r}
colnames(unfiltered_glowing_counts) <- gsub(x = colnames(unfiltered_glowing_counts),pattern = "\\.",replacement = "-")
rownames(unfiltered_glowing_counts) <- unfiltered_glowing_counts$Geneid


female_ymasked <- unfiltered_glowing_counts %>% select(contains("Ymasked"))
colnames(female_ymasked) <- str_extract(colnames(female_ymasked), "[^_]+")


default_alignment <- unfiltered_glowing_counts %>% select(contains("YPARsMasked"))
colnames(default_alignment) <- str_extract(colnames(default_alignment), "[^_]+")

male_yparsmasked <- default_alignment[colnames(default_alignment) %in% pheno[which(pheno$childs_sex=="Male"),"participant_id"]]
#I had the phenred codes...might be deleted ): 

sexalignment <- merge(female_ymasked,male_yparsmasked, by = 'row.names')
rownames(sexalignment) = sexalignment$Row.names
sexalignment =sexalignment[,-1]
```
## congruency between IDs in pheno and sex alignment 
```{r}
#which(colnames(sexalignment) %in% rownames(pheno) == FALSE)
#colnames(sexalignment)[12]
#G-165

#which(rownames(pheno) %in% colnames(sexalignment) == FALSE)
#rownames(pheno)[8]
#G-247
#rownames(pheno)[115]
#G-411
# rownames(pheno)[141] 
#G-316

sexalignment = sexalignment[,-12]
sexalignment = sexalignment[, order(colnames(sexalignment))]
pheno = pheno[-c(8,115,141),]
pheno = pheno[order(rownames(pheno)),]

#colnames(sexalignment) == rownames(pheno) 
#TRUE 
```
## create DESEQ data seq and counts dataset 
```{r}
deseq_sexalignment = DESeqDataSetFromMatrix(countData = sexalignment, colData = pheno, design = ~ 1)
nrow(deseq_sexalignment) # 56852
#rerrun this section 
keep <- rowSums(fpm(deseq_sexalignment)>=1) >= 75 #1 count per million (in data with ~20M reads: cpm 1 ~ 20 counts/sample)
deseq_sexalignment <- deseq_sexalignment[keep,]
nrow(deseq_sexalignment) # 16101

sex_complement_raw_counts = counts(deseq_sexalignment)

```
#identify genes on Y chr 
```{r}
library(biomaRt)
#library(cowplot)

ensembl=useMart("ensembl")
ensembl=useDataset("hsapiens_gene_ensembl",mart = ensembl)
ensembl.ids = colnames(combat.adj)
ensembl.genes <- biomaRt::getBM(attributes = c('hgnc_symbol', 'chromosome_name','start_position', 'end_position'), filters = 'hgnc_symbol', values = ensembl.ids, mart = ensembl)

Y_genes = ensembl.genes %>% filter(.$chromosome_name == "Y") %>% pull(hgnc_symbol)

Y_genes
#37 Y linked genes listed when  rowSums(fpm(deseq_sexalignment)>=1) >= 30 
#33 Y linked genes listed when rowSums(fpm(deseq_sexalignment)>=1) >= 75
```
#choose random genes to compare with original data 
```{r}
set.seed(110)
load("Glowing_RNAse_141QCd.RData")
GlowingPlacenta_RNAseq.rawCounts = counts
#dim(GlowingPlacenta_RNAseq.rawCounts)
#13831   141
#sum(rownames(GlowingPlacenta_RNAseq.rawCounts) %in% rownames(sex_complement_raw_counts) )
#11279
shared_genes = rownames(GlowingPlacenta_RNAseq.rawCounts)[which(rownames(GlowingPlacenta_RNAseq.rawCounts) %in% rownames(sex_complement_raw_counts) == TRUE)]
sample = sample(rownames(sex_complement_raw_counts[shared_genes,]),10)

#sample %in% rownames(GlowingPlacenta_RNAseq.rawCounts) 
#TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
```
#correlation matrix of random 10 genes from sample 
```{r}
cor_matrix <- cor(t(GlowingPlacenta_RNAseq.rawCounts[sample,]), t(sex_complement_raw_counts[sample,colnames(GlowingPlacenta_RNAseq.rawCounts)]))

cor_matrix
```
```{r}
plot(t(GlowingPlacenta_RNAseq.rawCounts["SGMS2",]),t(sex_complement_raw_counts["SGMS2",colnames(GlowingPlacenta_RNAseq.rawCounts)]), xlab = "original RNAseq raw counts",
     ylab = "Sex Complement RNAseq raw counts",main = "SGMS2")
```

```{r}
plot(t(GlowingPlacenta_RNAseq.rawCounts["XIST",]),t(sex_complement_raw_counts["XIST",colnames(GlowingPlacenta_RNAseq.rawCounts)]), xlab = "original RNAseq raw counts",
     ylab = "Sex Complement RNAseq raw counts",main = "XIST")
```

```{r}
#Y_genes %in% rownames(GlowingPlacenta_RNAseq.rawCounts)
 #FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE  TRUE FALSE FALSE FALSE  TRUE FALSE FALSE
 #FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE  TRUE  TRUE FALSE FALSE  TRUE

#sum( Y_genes %in% rownames(GlowingPlacenta_RNAseq.rawCounts)) #7 
colnamessexcomp = colnames(sex_complement_raw_counts)
sex_complement_raw_counts = data.frame(sex_complement_raw_counts)
colnames(sex_complement_raw_counts) = colnamessexcomp
Y_genes = Y_genes[which(Y_genes %in% rownames(GlowingPlacenta_RNAseq.rawCounts) == TRUE)]
for ( i in Y_genes) {
# Extract the raw counts for the current gene from GlowingPlacenta_RNAseq and sex_complement_raw_counts
  original_counts <- t(GlowingPlacenta_RNAseq.rawCounts[as.character(i), ])
  sex_complement_counts <- t(sex_complement_raw_counts[as.character(i), colnames(GlowingPlacenta_RNAseq.rawCounts)])
  
  # Create a scatter plot of the raw counts
  plot(original_counts,
       sex_complement_counts,
       col = pheno$childs_sex,
       xlab = "Original RNAseq raw counts",
       ylab = "Sex Complement RNAseq raw counts",
       main = as.character(i))
 model <- lm(sex_complement_counts ~ original_counts)
  r_squared <- summary(model)$r.squared
  
  # Add the R^2 value to the plot
  legend("topright", legend = paste("R^2 =", round(r_squared, 2)), bty = "n")
}


```

#check highly expressed genes that are shared between both datasets 
```{r}
max_sex_comp <- matrix(apply(sex_complement_raw_counts, 1, mean, na.rm=TRUE))
rownames(max_sex_comp) = rownames(sex_complement_raw_counts)
max_sex_comp = max_sex_comp[rownames(max_sex_comp)[order(max_sex_comp[,1],decreasing = T)],1]

head(max_sex_comp)

max_original <- matrix(apply(GlowingPlacenta_RNAseq.rawCounts, 1, mean, na.rm=TRUE))
rownames(max_original) = rownames(GlowingPlacenta_RNAseq.rawCounts)
max_original = max_original[rownames(max_original)[order(max_original[,1],decreasing = T)],1]

head(max_original)
```

```{r}
shared_genes = rownames(GlowingPlacenta_RNAseq.rawCounts)[which(rownames(GlowingPlacenta_RNAseq.rawCounts) %in% rownames(sex_complement_raw_counts) == TRUE)]

max_sex_comp <- matrix(apply(sex_complement_raw_counts[shared_genes,], 1, mean, na.rm=TRUE))
rownames(max_sex_comp) = shared_genes
max_sex_comp = max_sex_comp[rownames(max_sex_comp)[order(max_sex_comp[,1],decreasing = T)],1]

head(max_sex_comp)

max_original <- matrix(apply(GlowingPlacenta_RNAseq.rawCounts[shared_genes,], 1, mean, na.rm=TRUE))
rownames(max_original) = shared_genes
max_original = max_original[rownames(max_original)[order(max_original[,1],decreasing = T)],1]

head(max_original)

```
#RData file
```{r}
pheno_150 = pheno 
save(pheno_150, sex_complement_raw_counts, file = "sexcomplement_raw_counts_pheno150.RData")
```


